src/components/calcite-radio-button/calcite-radio-button.tsx

```diff
@@ -85,7 +89,12 @@ export class CalciteRadioButton {
 
   @Watch("name")
   nameChanged(newName: string): void {
-    this.input.name = newName;
+    if (this.name === newName) {
```

Comment:
Unbelievable right?  But rest assured I did see this fire in the webviewer app.  This is the real fix for this bug, since the `name` watcher was firing the second time it was rendered, it was executing the rest of this function and thus unchecking all the radios in the group because it thought the name actually changed when it didn't.  Very bizarre, and this could be another Stencil + Maquette bug too.  It wouldn't surprise me since I've seen all manner of unexpected things happening, from `connectedCallback` not firing the second time, and now this watcher firing even though the property it is watching didn't change.

Reply:
I just re-read your comment, and you mentioned "reference".  Perhaps that is why this watcher is firing despite the name being the same?  The reference to the DOM node in Maquette is new the second time around perhaps?